gtwrap/matlab_wrapper/wrapper.py

```diff
@@ -1269,9 +1269,9 @@ def generate_collector_function(self, func_id):
                     Collector_{class_name}::iterator item;
                     item = collector_{class_name}.find(self);
                     if(item != collector_{class_name}.end()) {{
-                      delete self;
                       collector_{class_name}.erase(item);
                     }}
```

Comment:
@varunagrawal Is there a reason we are keeping the collector in these lines of code? Or is this now obsolete code>